src/System.Security.Cryptography.Algorithms/tests/PaddingModeTests.cs

```diff
@@ -0,0 +1,115 @@
+
+            using (Aes a = Aes.Create())
+            {
+                a.Key = key;
+                a.IV = iv;
+                a.Mode = CipherMode.CBC;
+                a.Padding = paddingMode;
+
+                byte[] cipher = a.Encrypt(plainText);
+
+                // we cannot validate the cipher in this padding mode as it consists of random data
+                if (paddingMode != PaddingMode.ISO10126)
+                {
+                    Assert.Equal<byte>(expectedCipher, cipher);
+                }
+
+                // decrypt it with PaddingMode.None so that we can inspect the padding manually
+                a.Padding = PaddingMode.None;
+                byte[] decrypted = a.Decrypt(cipher);
+                ValidatePadding(decrypted, paddingMode, expectedPaddingSize);
+            }
+        }
+
+        private static void ValidatePadding(byte[] buffer, PaddingMode paddingMode, int expectedPaddingSize)
+        {
+            switch (paddingMode)
+            {
+                case PaddingMode.PKCS7:
+                    ValidatePKCS7Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.ANSIX923:
+                    ValidateANSIX923Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.ISO10126:
+                    ValidateISO10126Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.Zeros:
+                    ValidateZerosPadding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.None:
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private static void ValidateZerosPadding(byte[] buffer, int expectedPaddingSize)
+        {
+            for (int i = buffer.Length - 1; i > buffer.Length - 1 - expectedPaddingSize; i--)
+            {
+                Assert.Equal(0, buffer[i]);
+            }
+        }
+
+        private static void ValidatePKCS7Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            for (int i = buffer.Length - 1; i > buffer.Length - 1 - expectedPaddingSize; i--)
+            {
+                Assert.Equal(expectedPaddingSize, buffer[i]);
+            }
+        }
+
+        private static void ValidateANSIX923Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            Assert.Equal(buffer[buffer.Length - 1], expectedPaddingSize);
+
+            for (int i = buffer.Length - expectedPaddingSize; i < buffer.Length - 1; i++)
+            {
+                Assert.Equal(0, buffer[i]);
+            }
+        }
+
+        private static void ValidateISO10126Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            // there is nothing else to validate as all the other padding bytes are random.
+            Assert.Equal(buffer[buffer.Length - 1], expectedPaddingSize);
```

Comment:
```c#

if (expectedPaddingSize >= 5)
{
    bool hasNonZero = buffer.Skip(buffer.Length - expectedPaddingSize).Take(expectedPaddingSize - 1).Any(x => x != 0);
    Assert.True(hasNonZero, "ISO 10126 padding does not zero-fill");
}
```

Alternatively, for ISO10126 with expectedPaddingSize >= 5, encrypt it twice and check that the two buffers decrypt differently (under PaddingMode.None)